OpenRA.Game/Graphics/Viewport.cs

```diff
@@ -279,14 +279,26 @@ public CPos ViewToWorld(int2 view)
 		}
 
 		/// <summary>Returns an unfiltered list of all cells that could potentially contain the mouse cursor.</summary>
-		IEnumerable<MPos> CandidateMouseoverCells(int2 world)
+		public IEnumerable<MPos> CandidateMouseoverCells(int2 world)
 		{
 			var map = worldRenderer.World.Map;
+			var tileScale = map.Grid.TileScale / 2;
 			var minPos = worldRenderer.ProjectedPosition(world);
 
-			// Find all the cells that could potentially have been clicked
-			var a = map.CellContaining(minPos - new WVec(1024, 0, 0)).ToMPos(map.Grid.Type);
-			var b = map.CellContaining(minPos + new WVec(512, 512 * map.Grid.MaximumTerrainHeight, 0)).ToMPos(map.Grid.Type);
+			// Find all the cells that could potentially have been clicked.
+			MPos a;
+			MPos b;
+			if (map.Grid.Type == MapGridType.RectangularIsometric)
+			{
+				// TODO: this generates too many cells.
+				a = map.CellContaining(minPos - new WVec(tileScale, 0, 0)).ToMPos(map.Grid.Type);
```

Comment:
This was `1024` (== `2 * tileScale`) before. Intentional?

Reply:
yes, and it's still too large of a number